.github/workflows/run-e2e-on-fork.yml

```diff
@@ -0,0 +1,150 @@
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              username: '${{ github.event.sender.login }}'
+            })
+            return data.permission;
+      - name: 'Fail this workflow if user does not have write permissions'
+        if: ${{ steps.get-permissions.outputs.result != 'write' && steps.get-permissions.outputs.result != 'admin' }}
+        run: |
+          echo "${{ steps.get-permissions.outputs.result }}"
+          exit 1
+  setup:
+    needs: check-write-access
+    runs-on: ubuntu-latest
+    outputs:
+      commit_id: ${{ steps.read-artifact.outputs.result }}
+    steps:
+      - name: 'Download artifact'
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975
+        with:
+          script: |
+            const artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{github.event.workflow_run.id }},
+            });
+            const matchArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "commit"
+            })[0];
+            const download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            const fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/commit.zip', Buffer.from(download.data));
+      - run: unzip commit.zip
+      - name: 'Read from artifact zip'
+        uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975
+        id: read-artifact
+        with:
+          result-encoding: string
+          script: |
+            const fs = require('fs');
+            const commit_id = fs.readFileSync('./commit_id', 'utf-8');
+            return commit_id
+              .replace(/(\r\n|\n|\r)/gm, '') // remove last new line character
+              .replace(/[^A-Za-z0-9]/g, ''); // remove non-alphanumeric characters
+      # create status
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c #v3.3.0 https://github.com/actions/checkout/commit/ac593985615ec2ede58e132d2e21d2b1cbd6127c
+      - name: Set status to commit sha
+        uses: ./.github/actions/set-status
+        with:
+          sha: ${{ steps.read-artifact.outputs.result }}
+          state: 'pending'
+          context: 'Run E2E tests'
+          description: 'E2E tests are now running'
+          # URL below is a link to the current workflow run to allow users to see the status of the workflow.
+          target-url: https://github.com/${{ github.event.repository.owner.login }}/${{ github.event.repository.name }}/actions/runs/${{ github.run_id }}
+
+  setup-cache:
+    needs: setup
+    uses: wlee221/amplify-ui/.github/workflows/reusable-setup-cache.yml@main
```

Comment:
`wlee221`? Is this intentional? same comment for bellow

Reply:
oh will update this

Reply:
🫠